lib/temporal/client.rb

```diff
@@ -409,6 +409,10 @@ def query_workflow_executions(namespace, query, next_page_token: nil, max_page_s
       Temporal::Workflow::Executions.new(connection: connection, status: :all, request_options: { namespace: namespace, query: query, next_page_token: next_page_token, max_page_size: max_page_size }.merge(filter))
     end
 
+    def count_workflow_executions(namespace, query)
```

Comment:
I don't think that other SDKs actually have this method exposed. With that in mind, let's not add it to the `Client`, but instead only expose it on the connection itself and possibly make the connection public. This way you can access the low-level API if needed, but we're not committing to supporting it via the high-level classes